### PR TITLE
First draft of models

### PR DESCRIPTION
**Pull request by Keats**

### CODE REVIEW

File: oven/foods/models.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+from model_utils.models import TimeStampedModel
+
+
+class Food(TimeStampedModel):
```
Comment (carlos-alberto) [inline oven/foods/models.py:6]: I would prefer the name ingredient throughout.

File: oven/recipes/models.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    text = models.TextField()
+
+    # Which step of the recipe this is
+    number = models.IntegerField()
+
+    def __str__(self):
+        return 'Step %d for %s' % (self.number, self.recipe.title)
+
+
+class StepPicture(TimeStampedModel):
```
Comment (deane) [inline oven/recipes/models.py]: StepPicture and Recipe picture seem very similar. Why not a Picture model, with a many to one(/many?) on Recipe and Step?

File: oven/foods/models.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+from model_utils.models import TimeStampedModel
```
Comment (deane) [inline oven/foods/models.py:3]: Where is model_utils? why does the Food model has to be timestamped?

File: oven/recipes/models.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class Ingredient(TimeStampedModel):
+    recipe = models.ForeignKey(Recipe)
+    food = models.ForeignKey(Food)
+
+    # In which order does it appear in the ingredient list
+    order = models.IntegerField()
+
+    # TODO: this won't work with units other than grams
```
Comment (deane) [inline oven/recipes/models.py]: I see 2 solutions here:
1 - storing it as a string (something like "250g", "180ml" or "3tbsp")
2 - add a field (char field) for the unit

File: oven/recipes/models.py (hunk 2 above)
Comment (Keats) [inline oven/recipes/models.py]: Yep I thought the same as I was doing it

File: oven/foods/models.py (hunk 3 above)
Comment (Keats) [inline oven/foods/models.py:3]: comes from django-model-utils

It's just easier to timestamp everything

File: oven/recipes/models.py (hunk 4 above)
Comment (Keats) [inline oven/recipes/models.py]: we need a way to get the quantity in gram for the macro though

File: oven/foods/models.py (hunk 3 above)
Comment (deane) [inline oven/foods/models.py:3]: Timestamp the food model doesn't seem to make much sense, but I don't mind it being there

File: oven/recipes/models.py (hunk 4 above)
Comment (deane) [inline oven/recipes/models.py]: idea: Add a type on food ('solid', 'liquid', 'powder', others if we ever need)
`get_macro(food, quantity, unit)` util function, and using constants like `{'tbsp': {'liquid': 10, 'powder': 3}, 'pinch': {'powder': 1}, 'pound': {'solid':...` to get macros given a food, quantity and a unit

Comment (Keats) [inline oven/recipes/models.py]: hum another problem is for example eggs.
You count them as a unit, how do you convert 2 eggs in grams ?

Comment (deane) [inline oven/recipes/models.py]: you could have "egg" as a unit for the egg food. Should be just a few cases like this

Comment (carlos-alberto) [inline oven/recipes/models.py]: Yeah I like this idea. Allows easy conversions between standard units, but easy for users to understand.

File: oven/recipes/models.py (hunk 2 above)
Comment (carlos-alberto) [inline oven/recipes/models.py]: +1

File: oven/recipes/models.py (hunk 4 above)
Comment (Keats) [inline oven/recipes/models.py]: that would be unit instead of the name of the food.
Either way I will need to modify the food db then

**Comment by carlos-alberto** [comment]
ship it

**Pull request merged**